### PR TITLE
Adding new Data Points

### PR DESCRIPTION
**Pull request by khilangoel**
Adding Telemetry Data points

### CODE REVIEW

File: src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs [hunk 1]
```diff
@@ -227,10 +227,35 @@ private int VerifyParallelSettingAndCalculateParallelLevel(int sourceCount, stri
             return parallelLevelToUse;
         }
 
-        private bool ShouldRunInNoIsolation(string runsettings, bool isParallelEnabled, bool isDataCollectorEnabled)
+        private bool ShouldRunInNoIsolation(string runsettings, bool isParallelEnabled, bool isDataCollectorEnabled, IRequestData requestData)
         {
             var runConfiguration = XmlRunSettingsUtilities.GetRunConfigurationNode(runsettings);
 
+            // Collecting Target Framework.
+            requestData.MetricsCollection.Add(TelemetryDataConstants.TargetFramework, runConfiguration.TargetFrameworkVersion);
```
Comment (codito) [inline src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs]: Why is it the responsibility of `ShouldRunInIsolation` to collect metrics data?

**Review by codito — APPROVED** [review]

**Comment by khilan2goel** [comment]
@dotnet-bot  test Windows_NT / Debug Build

**Comment by khilan2goel** [comment]
@dotnet-bot test Windows_NT / Release Build

**Comment by khilan2goel** [comment]
@dotnet-bot  please test this

**Review by harshjain2 — APPROVED** [review]

**Comment by khilan2goel** [comment]
@dotnet-bot test this please

**Pull request merged**